libs/core/tests/unit/mutex.cpp

```diff
@@ -22,56 +22,84 @@
 
 #include <chrono>
 #include <cstdint>
+#include <functional>
 #include <thread>
 #include <vector>
 
-#ifdef FETCH_DEBUG_MUTEX
+using Mutex  = fetch::SimpleDebugMutex;
+using RMutex = fetch::RecursiveDebugMutex;
+
+using namespace std::chrono_literals;
 
 TEST(DebugMutex, SimpleProblem)
 {
-  fetch::MutexRegister::ThrowOnDeadlock();
+  fetch::DeadlockHandler::ThrowOnDeadlock();
   {
-    fetch::DebugMutex                  mutex;
-    std::lock_guard<fetch::DebugMutex> guard1(mutex);
-    EXPECT_THROW(std::lock_guard<fetch::DebugMutex> guard2(mutex), std::runtime_error);
+    Mutex                  mutex;
+    std::lock_guard<Mutex> guard1(mutex);
+    EXPECT_THROW(std::lock_guard<Mutex> guard2(mutex), std::runtime_error);
   }
 
+  static constexpr std::size_t TABLE_SIZE = 7;
```

Comment:
What is the dining philosophers testing here? It seems racy, if you change the logic to this:

```
  for (std::size_t fork = 0; fork < TABLE_SIZE; ++fork)
  {
    Mutex &left_fork  = forks[fork];
    Mutex &right_fork = forks[(fork + 1) % TABLE_SIZE];
    dining_philosophers.emplace_back(runnable, std::ref(left_fork), std::ref(right_fork));
    std::this_thread::sleep_for(std::chrono::milliseconds(95));
  }
```

It will likely fail I believe